Yosemite/Yosemite/PointOfSale/POSProductProvider.swift

```diff
@@ -66,20 +67,31 @@ final class POSProductProvider {
 // MARK: - PreviewProvider helpers
 //
 extension POSProductProvider {
-    static func provideProductForPreview(currencySettings: CurrencySettings = ServiceLocator.currencySettings) -> POSProduct {
+    public static func provideProductForPreview() -> POSProduct {
         POSProduct(itemID: UUID(),
                    productID: 1,
                    name: "Product 1",
                    price: "1.00",
-                   currencySettings: currencySettings)
+                   formattedPrice: "$1.00")
     }
 
-    static func provideProductsForPreview(currencySettings: CurrencySettings = ServiceLocator.currencySettings) -> [POSProduct] {
+    public static func provideProductsForPreview() -> [POSProduct] {
         return [
-            POSProduct(itemID: UUID(), productID: 1, name: "Product 1", price: "1.00", currencySettings: currencySettings),
-            POSProduct(itemID: UUID(), productID: 2, name: "Product 2", price: "2.00", currencySettings: currencySettings),
-            POSProduct(itemID: UUID(), productID: 3, name: "Product 3", price: "3.00", currencySettings: currencySettings),
-            POSProduct(itemID: UUID(), productID: 4, name: "Product 4", price: "4.00", currencySettings: currencySettings),
+            POSProduct(itemID: UUID(), productID: 1, name: "Product 1", price: "1.00", formattedPrice: "$1.00"),
+            POSProduct(itemID: UUID(), productID: 2, name: "Product 2", price: "2.00", formattedPrice: "$2.00"),
+            POSProduct(itemID: UUID(), productID: 3, name: "Product 3", price: "3.00", formattedPrice: "$3.00"),
+            POSProduct(itemID: UUID(), productID: 4, name: "Product 4", price: "4.00", formattedPrice: "$4.00"),
         ]
     }
 }
+
+/// Null product provider that acts as preview helper for the Point of Sale feature
+///
+public final class NullPOSProductProvider: POSItemProvider {
```

Comment:
nit: since this seems to be used only for SwiftUI previews within DEBUG, how about moving this to the app layer within DEBUG?

Reply:
Agree, I've added it to the same issue as the Yosemite dependencies for previews only. Will be tackled separately here: https://github.com/woocommerce/woocommerce-ios/issues/12917